### PR TITLE
Initial alias linter tool

### PR DESCRIPTION
**Pull request by kminder**
Initial version of a tool to detect doc pages in prior releases that have been removed but do not have aliases in newer releases to prevent not found errors.
Usage
`go run tools/alias_linter/main.go -priorBranches "release-1.0,release-1.1,release-1.2,release-1.3,release-1.4" -checkBranch release-1.5`
See README.md for more details.

### CODE REVIEW

File: tools/alias_linter/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# or between two specific release branches
+go run tools/alias_linter/main.go -priorBranches release-1.4 -checkBranch release-1.5
+```
+
+This will output a list of the pages of the prior branches that are missing in the check branch.
+If there are no missing pages the tool will exit with a 0 exit code.
+Otherwise the tool will exit with a 1 exit code.
+
+By default the tool does an in memory clone of the https://github.com/verrazzano/docs.git repository.
+This can be time consuming an may eventually be tool large.
```
Comment (bobdonat) [inline tools/alias_linter/README.md:19]: typo "tool"

File: tools/alias_linter/main.go [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+	if len(args.repoDir) == 0 && len(args.repoURL) == 0 {
+		return nil, fmt.Errorf("must provide either repoURL or repoDir")
+	}
+	if len(args.repoDir) > 0 {
+		return openRepoOnDisk(args.repoDir)
+	}
+	if len(args.repoURL) > 0 {
+		return cloneRepoInMemory(args.repoURL)
+	}
+	return nil, fmt.Errorf("invalid repoURL and repoDir combination")
```
Comment (bobdonat) [inline tools/alias_linter/main.go:112]: I think the intention was for this error to be returned when both Dir and URL are specified.  Looks like the check has to be done earlier.

File: tools/alias_linter/main.go [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+func convertFileNameToPageName(fileName string) string {
+	fileName = strings.TrimSuffix(fileName, "/_index.md")
+	m := markdown_re.FindStringSubmatch(fileName)
+	if m != nil {
+		fileName = m[1]
+	}
+	if len(fileName) > 1 {
+		fileName = strings.TrimPrefix(fileName, "/")
+	}
+	if len(fileName) > 1 {
```
Comment (bobdonat) [inline tools/alias_linter/main.go:179]: is this if statement intended to be repeated

Comment (kminder) [reply tools/alias_linter/main.go:177]: Fixed

File: tools/alias_linter/main.go (hunk 2 above)
Comment (kminder) [reply tools/alias_linter/main.go]: Fixed

**Review by bobdonat — APPROVED** [review]

**Pull request merged**